activesupport/lib/active_support/core_ext/object/json.rb

```diff
@@ -165,7 +165,8 @@ def as_json(options = nil) # :nodoc:
 class Array
   def as_json(options = nil) # :nodoc:
     if options
-      map { |v| v.as_json(options.dup) }
+      options = options.dup.freeze unless options.frozen?
```

Comment:
unfortunately the `freeze` call  breaks active_model_serializers here : https://github.com/rails-api/active_model_serializers/blob/bda195020e77f9fd4850d4beeb3d3e51bfa086e6/lib/active_model_serializers/adapter/attributes.rb#L13

and probably some gems that manipulate /override rails `options`

@jhawthorn should i patch activemodelserializers or should I create an issue here, sorry for the ping but I'm not sure.

Probably the fix is simply to call `options.dup` in AMS the same way you did here I think

Anyways, I just wanted to inform you!

Reply:
Thanks for the ping! I'm open to reverting this if compatibility is going to be too much of a challenge, but it's really hard to throw away a 2x improvement to something as common as `to_json`.

Either way let's patch ActiveModelSerializers. IMO modifying an options hash passed in is a bug.

Reply:
@chaadow patching AMS would be better at this point. I have this issue too.